it/kafka/src/test/java/org/apache/beam/it/kafka/KafkaIOST.java

```diff
@@ -0,0 +1,480 @@
+        org.joda.time.Duration.standardMinutes(configuration.minutes).getMillis();
+    long totalRows = startMultiplier * stopAfterMillis / fireInterval;
+    List<LoadPeriod> loadPeriods =
+        getLoadPeriods(configuration.minutes, DEFAULT_LOAD_INCREASE_ARRAY);
+
+    PCollection<byte[]> source =
+        writePipeline
+            .apply(
+                PeriodicImpulse.create()
+                    .stopAfter(org.joda.time.Duration.millis(stopAfterMillis - 1))
+                    .withInterval(org.joda.time.Duration.millis(fireInterval)))
+            .apply(
+                "Extract values",
+                MapElements.into(TypeDescriptor.of(byte[].class))
+                    .via(instant -> Longs.toByteArray(instant.getMillis() % totalRows)));
+    if (startMultiplier > 1) {
+      source =
+          source
+              .apply(
+                  "One input to multiple outputs",
+                  ParDo.of(new MultiplierDoFn(startMultiplier, loadPeriods)))
+              .apply("Reshuffle fanout", Reshuffle.viaRandomKey())
+              .apply("Counting element", ParDo.of(new CountingFn<>(WRITE_ELEMENT_METRIC_NAME)));
+    }
+    source.apply(
+        "Write to Kafka",
+        KafkaIO.<byte[], byte[]>write()
+            .withBootstrapServers(configuration.bootstrapServers)
+            .withTopic(kafkaTopic)
+            .withValueSerializer(ByteArraySerializer.class)
+            .withProducerConfigUpdates(
+                ImmutableMap.of(
+                    ProducerConfig.RETRIES_CONFIG, 10,
+                    ProducerConfig.REQUEST_TIMEOUT_MS_CONFIG, 600000,
+                    ProducerConfig.RETRY_BACKOFF_MS_CONFIG, 5000))
+            .values());
+
+    PipelineLauncher.LaunchConfig options =
+        PipelineLauncher.LaunchConfig.builder("write-kafka")
+            .setSdk(PipelineLauncher.Sdk.JAVA)
+            .setPipeline(writePipeline)
+            .addParameter("runner", configuration.runner)
+            .addParameter(
+                "autoscalingAlgorithm",
+                DataflowPipelineWorkerPoolOptions.AutoscalingAlgorithmType.THROUGHPUT_BASED
+                    .toString())
+            .addParameter("numWorkers", String.valueOf(configuration.numWorkers))
+            .addParameter("maxNumWorkers", String.valueOf(configuration.maxNumWorkers))
+            .addParameter("experiments", "use_runner_v2")
+            .build();
+
+    return pipelineLauncher.launch(project, region, options);
+  }
+
+  /** The method reads data from Kafka topic in streaming mode. */
+  private PipelineLauncher.LaunchInfo readData() throws IOException {
+    KafkaIO.Read<byte[], byte[]> readFromKafka =
+        KafkaIO.readBytes()
+            .withBootstrapServers(configuration.bootstrapServers)
+            .withTopic(kafkaTopic)
+            .withConsumerConfigUpdates(ImmutableMap.of("auto.offset.reset", "earliest"));
+
+    readPipeline
+        .apply("Read from unbounded Kafka", readFromKafka)
```

Comment:
minor - the transform naming is not quite accurate. Basically Kafka read has two modes (unbounded or SDF) and the default is runner dependent.

e.g. Dataflow legacy runner will choose unbounded kafka; runner v2 will choose SDF. And this can be overriden by pipeline option.

Ideally one can configure the read mode but it's good for now